File: 0901-online-stock-span/0901-online-stock-span.java
class StockSpanner {
    private Stack<int[]> stack;
    public StockSpanner() {
        this.stack = new Stack<>();
    }

    public int next(int price) {
        int count = 1;
        while(!stack.isEmpty() && stack.peek()[0] <= price){
            count += stack.pop()[1];
        }
        stack.push(new int[]{price, count});
        return count;
    }
}

File: 901-online-stock-span/0901-online-stock-span.cpp
 class StockSpanner {
 public:
 	StockSpanner() {}		
 	int next(int price) {
 		int span = 1;
 		while(!stack_.empty() && price >= stack_.top().first){
 			span += stack_.top().second;
 			stack_.pop();
 		}
 		stack_.emplace(price, span);
 		return span;
 	}
 private:
 	stack<pair<int, int>> stack_; // first: price, second: span
 };